torch/_inductor/fx_passes/b2b_gemm.py

```diff
@@ -445,39 +527,32 @@ def all_reach_via_pointwise_with_no_other_inputs(
         with no other input nodes for the intermediates and dst;
         return
         (1) the Boolean value
-        (2) the subgraph node set (which only makes sense when the Boolean value is True)
+        (2) the subgraph node set including src and dst (which only makes sense when the Boolean value is True)
         """
-        if src == dst:  # trivial subgraph
-            return (True, {src})
-
-        reachable: Dict[torch.fx.Node, bool] = {}
+        visited: Set[torch.fx.Node] = set()
         input_counter: Dict[torch.fx.Node, int] = {}
 
-        def reach(node: torch.fx.Node) -> bool:
-            if node in reachable:
-                return reachable[node]
-            if node is dst:
-                reachable[node] = True
-                return True
-            # for nodes other than dst, bookkeep their users' input counts
-            for user in node.users.keys():
-                if user not in input_counter:
-                    input_counter[user] = len(user.all_input_nodes)
-                input_counter[user] -= 1
-            # continue checking reachability
-            if (node is src) or is_pointwise_node(node):
-                ret = True
-                for user in node.users.keys():
-                    if not reach(user):
-                        ret = False
-                        break
-            else:
-                ret = False
-            reachable[node] = ret
-            return ret
-
-        ok = reach(src) and all(count == 0 for count in input_counter.values())
-        return (ok, set(reachable.keys()))
+        all_reachable = True
+        queue = deque([src])
+        while queue:
+            node = queue.popleft()
+            if node not in visited:
```

Comment:
very small nit, but something like this organization would be slightly more readable to my eyes:

```
if node not in visited:
    visited.add(node)
    if (node is src) or is_pointwise_node(node):
        ...
    elif not node is dst:
        all_reachable = False
        break
````